pkgs/stdenv/generic/check-meta.nix

```
@@ -86,12 +89,41 @@ let
     allowInsecurePredicate attrs ||
     builtins.getEnv "NIXPKGS_ALLOW_INSECURE" == "1";
 
-  showLicense = license: toString (map (l: l.shortName or "unknown") (lib.lists.toList license));
+  hasSourceProvenance = attrs:
+    attrs ? meta.sourceProvenance;
+
+  isNonSource = sourceTypes: lib.lists.any (t: !t.isSource or true) sourceTypes;
```

```suggestion
  isNonSource = sourceTypes: lib.lists.any (t: !t.isSource) sourceTypes;
```

The `or true` should be superfluous by our construction of fields of `lib.sourceTypes`.

Using `str` source types, this introduces a bug as `"binaryNativeCode".isSource or true` evaluates to `true`. (See `str` source types below.)